os/arch/arm/src/armv7-a/Toolchain.defs

```diff
@@ -0,0 +1,231 @@
+endif
+
+ifeq ($(filter y, \
+      $(CONFIG_ARMV7A_TOOLCHAIN_GNU_EABIW) \
+    ),y)
+  CONFIG_ARMV7A_TOOLCHAIN ?= GNU_EABIW
+endif
+
+ifeq ($(filter y, \
+      $(CONFIG_ARMV7A_TOOLCHAIN_CLANGL) \
+    ),y)
+  CONFIG_ARMV7A_TOOLCHAIN ?= CLANGL
+endif
+
+ifeq ($(filter y, \
+      $(CONFIG_ARMV7A_TOOLCHAIN_CLANGW) \
+    ),y)
+  CONFIG_ARMV7A_TOOLCHAIN ?= CLANGW
+endif
+
+#
+# Supported toolchains
+#
+# TODO - It's likely that all of these toolchains now support the
+#        CortexM4.  Since they are all GCC-based, we could almost
+#        certainly simplify this further.
+#
+# Each toolchain definition should set:
+#
+#  CROSSDEV         The GNU toolchain triple (command prefix)
+#  ARCROSSDEV       If required, an alternative prefix used when
+#                   invoking ar and nm.
+#  ARCHCPUFLAGS     CPU-specific flags selecting the instruction set
+#                   FPU options, etc.
+#  MAXOPTIMIZATION  The maximum optimization level that results in
+#                   reliable code generation.
+#
+
+ifeq ($(CONFIG_DEBUG_CUSTOMOPT),y)
+  MAXOPTIMIZATION := $(CONFIG_DEBUG_OPTLEVEL)
+endif
+
+ifeq ($(CONFIG_ARCH_CORTEXA5),y)
+  ARCHCPUFLAGS += -mcpu=cortex-a5
+  ZARCHCPUFLAGS := -mcpu=cortex_a5
+else ifeq ($(CONFIG_ARCH_CORTEXA7),y)
+  ARCHCPUFLAGS += -mcpu=cortex-a7
+  ZARCHCPUFLAGS := -mcpu=cortex_a7
+else ifeq ($(CONFIG_ARCH_CORTEXA8),y)
+  ARCHCPUFLAGS += -mcpu=cortex-a8
+  ZARCHCPUFLAGS := -mcpu=cortex_a8
+else ifeq ($(CONFIG_ARCH_CORTEXA9),y)
+  ARCHCPUFLAGS += -mcpu=cortex-a9
+  ZARCHCPUFLAGS := -mcpu=cortex_a9
+else ifeq ($(CONFIG_ARCH_CORTEXA32),y)
+  ARCHCPUFLAGS += 
+  ZARCHCPUFLAGS := -mcpu=cortex_a32
+endif
+
+ifneq ($(CONFIG_ARM_DPFPU32),y)
+  ARCHFPUD16 = -d16
+endif
+
+# Cortex-A5  | -mfpu=vfpv4-fp16 | -mfpu=vfpv4-d16-fp16 | -mfpu=neon-fp16
+# Cortex-A7  | -mfpu=vfpv4      | -mfpu=vfpv4-d16      | -mfpu=neon-vfpv4
+# Cortex-A8  | -mfpu=vfpv3      |                      | -mfpu=neon (alias for neon-vfpv3)
+# Cortex-A9  | -mfpu=vfpv3-fp16 | -mfpu=vfpv3-d16-fp16 | -mfpu=neon-fp16
+# Cortex-A15 | -mfpu=vfpv4      |                      | -mfpu=neon-vfpv4
+
+ifeq ($(CONFIG_ARCH_FPU),y)
+  ifeq ($(CONFIG_ARM_FPU_ABI_SOFT),y)
+    ARCHCPUFLAGS += -mfloat-abi=softfp
+  else
+    ARCHCPUFLAGS += -mfloat-abi=hard
+  endif
+
+  ifeq ($(CONFIG_ARM_NEON),y)
+    ARCHNEON = neon-
+  endif
+  ifeq ($(CONFIG_ARCH_CORTEXA8),y)
+    ARCHFPU = vfpv3
+  else ifeq ($(CONFIG_ARCH_CORTEXA32),y)
+    ARCHFPU = vfpv3
+  else ifeq ($(CONFIG_ARCH_CORTEXA9),y)
+    ARCHFPU = vfpv3
+  else
+    ARCHFPU = vfpv4
+  endif
+  ARCHCPUFLAGS += -mfpu=$(ARCHNEON)$(ARCHFPU)$(ARCHFPUD16)
+
+else
+  ARCHCPUFLAGS += -mfloat-abi=soft
+endif
+
+ifeq ($(CONFIG_ARM_TOOLCHAIN_ARMCLANG),y)
+
+  ifeq ($(CONFIG_ARCH_CORTEXA5),y)
+    LDCPUFLAGS = Cortex-A5
+  else ifeq ($(CONFIG_ARCH_CORTEXA7),y)
+    LDCPUFLAGS = Cortex-A7
+  else ifeq ($(CONFIG_ARCH_CORTEXA8),y)
+    LDCPUFLAGS = Cortex-A8
+#Added 
+  else ifeq ($(CONFIG_ARCH_CORTEXA32),y)
```

Comment:
also, are CONFIG_ARCH_CORTEXA* added in the respective Kconfig? I think we will need to add them in arch/arm/Kconfig . Please check if the file was missed in the commit.

Reply:
It is added in the next commit. This commit has changes related to src/armv7a include/armv7a